builder/dockerfile/internals.go

```diff
@@ -569,8 +575,13 @@ func (b *Builder) run(cID string) (err error) {
 			Code:    ret,
 		}
 	}
-
-	return nil
+	once.Do(finish)
```

Comment:
Why do you need this `once.Do`, since you already deferred it? Why do you need the `sync.Once` at all?